lib/plugins/aws/deploy/lib/createStack.js

```diff
@@ -63,6 +60,13 @@ module.exports = {
           .then(() => BbPromise.resolve('alreadyCreated'))
           .catch(e => {
             if (e.message.indexOf('does not exist') > -1) {
+              if (this.serverless.service.provider.deploymentBucket) {
+                this.createLater = true;
+              }
+
+              if (this.options.noDeploy || this.createLater) {
```

Comment:
because of that we need that `this.createLater` flag after checking that the stack doesn't exist, if `noDeploy` was out of the catch block, this flag wouldn't be available, and it can only be available after checking AWS